dnf/util.py

```diff
@@ -644,29 +644,51 @@ def _is_file_pattern_present(specs):
 
 
 def is_container():
+    """Returns true if DNF is running in a container,
+       false otherwise.
+    """
+    # This attempts to reimplement
+    # https://github.com/ostreedev/ostree-rs-ext/blob/b0d1c1e4748c8708149bb6c92998da623260c4c6/lib/src/container_utils.rs#L20
+    # in python.
+    container_env_var = 'container'
+    container_env_file = '/run/.containerenv'
+    docker_env_file =  '/.dockerenv'
+    # https://stackoverflow.com/questions/20010199/how-to-determine-if-a-process-runs-inside-lxc-docker
+    if os.getenv(container_env_var) is not None or os.path.isfile(docker_env_file) or os.path.isfile(container_env_file):
+        return True
+    else:
+        return False
+
+def is_bootc_host():
     """Returns true is the system is managed as an immutable container,
        false otherwise.  If msg is True, a warning message is displayed
        for the user.
     """
 
-    bootc = '/usr/bin/bootc'
-    ostree = '/sysroot/ostree'
-
-    if os.path.isfile(bootc) and os.access(bootc, os.X_OK):
-        p = subprocess.Popen([bootc, "status", "--json"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        (out, err) = p.communicate()
-
-        if p.returncode == 0:
-            # check the output of 'bootc status'
-            j = json.loads(out)
+    ostree_booted = '/run/ostree-booted'
+    usr = '/usr/'
 
-            # XXX: the API from bootc status is evolving
-            status = j.get("status", "")
-            kind = j.get("kind", "")
+    # Check if usr is writtable and we are in a running ostree system.
+    # We want this code to return true only when the system is in locked state. If someone ran
+    # bootc overlay or ostree admin unlock we would want normal DNF path to be ran as it will be
+    # temporary changes (until reboot).
+    if os.path.isfile(ostree_booted) and not os.access(usr, os.X_OK):
```

Comment:
Did you mean `W_OK`?

Reply:
wow, yeah thanks for catching that! I copied the access check this code had before for bootc and did not update this to check for Writable instead of executable. :/